clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -1272,6 +1313,8 @@ Expected<StringRef> clang(ArrayRef<StringRef> InputFiles, const ArgList &Args) {
   if (!Triple.isNVPTX())
     CmdArgs.push_back("-Wl,--no-undefined");
 
+  if (IsSYCLKind && Triple.isNVPTX())
```

Comment:
do we know why this is SYCL specific and not NVPTX specific?

Reply:
For now, I am trying to match the behavior of new offloading model with that found in the new offloading model (for SYCL). This particular flow (getting the assembly file from the clang call and then explicitly calling ptxas inside the linker-wrapper tool) matches with what is found in the old offloading model, but does not match the community flow. Hence the SYCL specificity. This surely warrants revisiting at a later time.